Src/App/Router.php

```diff
@@ -4,35 +4,30 @@
 
 class Router
 {
+    private $interfaceControllerPath = '\Aot\Controller\ControllerInterface';
 
     /**
      * lead to the controller
      * @param $parameters
-     * @return mixed
+     * @return array
      * @throws \Exception
      */
     public function route($parameters)
     {
-
-        if (isset($parameters['controller']) && isset($parameters['method'])) {
-            $controllerClassName = ucfirst($parameters['controller']) . 'Controller';
-            $controllerName = '\Aot\Controller\\' . $controllerClassName;
-            $method = $parameters['method'];
-        } else {
+        if (!isset($parameters['controller']) || !isset($parameters['method'])) {
             throw new \Exception("method or controller invalid");
         }
-        if (class_exists($controllerName)) {
-            $controller = new $controllerName();
-        } else {
+        $controllerClassName = ucfirst($parameters['controller']) . 'Controller';
+        $controllerName = '\Aot\Controller\\' . $controllerClassName;
+        $method = $parameters['method'];
+        if (!class_exists($controllerName)) {
             throw new \Exception("controller invalid");
         }
-        if (is_callable($method, $controllerClassName)) {
-            unset($parameters['method'], $parameters['controller']);
-            return $controller->$method($parameters);
-            } else {
-            throw new \Exception("method invalid");
+        $controller = new $controllerName();
+        unset($parameters['method'], $parameters['controller']);
+        if (!is_a($controller, $this->interfaceControllerPath)) {
```

Comment:
with a `true` as third parameter you can remove the `class_exists` check

Reply:
done